Rubberduck.Inspections/Concrete/EmptyConditionBlockInspection.cs

```diff
@@ -29,33 +46,49 @@ public override IEnumerable<IInspectionResult> GetInspectionResults()
                                                         result));
         }
 
-        public override IInspectionListener Listener { get; } =
-            new EmptyConditionBlockListener();
+        public override IInspectionListener Listener { get; } = new EmptyConditionBlockListener(_blockToInspect);
 
         public class EmptyConditionBlockListener : EmptyBlockInspectionListenerBase
         {
+            ConditionBlockToInspect _blockToInspect;
+
+            public EmptyConditionBlockListener(ConditionBlockToInspect blockToInspect)
+            {
+                _blockToInspect = blockToInspect;
+            }
+
             public override void EnterIfStmt([NotNull] VBAParser.IfStmtContext context)
             {
-                InspectBlockForExecutableStatements(context.block(), context);
+                if ((_blockToInspect & ConditionBlockToInspect.If) == ConditionBlockToInspect.If)
```

Comment:
use `.HasFlag` here

Reply:
Much cleaner. Couldn't understand it last night.